### PR TITLE
Fix arm64/Windows build.

### PR DESCRIPTION
**Pull request by bruvzg**
- Fixes typo in the PNG driver SCsub
- Disables unsupported `raycast/embree` for arm64/Windows.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: modules/raycast/config.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 def can_build(env, platform):
+    # arm64/MSVC is not supported for now.
```
Comment (kus04e4ek) [inline modules/raycast/config.py:2]:
```suggestion
    # arm64/Windows is not supported for now.
```

**Review by fire — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Might be worth checking if some of these changes are needed for `3.x` too. I suspect at least the raycast one, but there might be more.

**Comment by aaronfranke** [comment]
Note: We can re-enable Embree if we apply this patch: https://github.com/RenderKit/embree/pull/483

I did not submit it here yet because I was hoping it would be accepted by upstream in a timely manner.

**Comment by akien-mga** [comment]
Looks pretty good to me. I think we can include it downstream, together with a patch file in `thirdparty/embree/patches/` that documents the downstream changes.

**Comment by aaronfranke** [comment]
Here's a branch with a patch file: https://github.com/aaronfranke/godot/tree/embree-win-arm64-msvc

Unfortunately it fails linking, I'm not sure what to do to fix it. @bruvzg Want to take a look?

**Comment by bruvzg** [comment]
> Unfortunately it fails linking, I'm not sure what to do to fix it.

I'll check it.

**Comment by bruvzg** [comment]
> Note: We can re-enable Embree if we apply this patch

I do not think it will work. `USE_SOFT_INTRINSICS` seems to only do something if you use `softintrin.h` (or generic `intrin.h`) instead of `*mmintrin.h`, and it does not have all required stuff implemented.

**Pull request merged**